pkg/process/procutil/process_linux.go

```diff
@@ -496,77 +508,77 @@ func (p *probe) parseStatusLine(line []byte, sInfo *statusInfo) {
 		if i+2 < len(line) && line[i] == ':' && unicode.IsSpace(rune(line[i+1])) {
 			key := line[0:i]
 			value := line[i+2:]
-			p.parseStatusKV(string(key), string(value), sInfo)
+			p.parseStatusKV(key, value, sInfo)
 			break
 		}
 	}
 }
 
 // parseStatusKV takes tokens parsed from each line in "status" file and populates statusInfo object
-func (p *probe) parseStatusKV(key, value string, sInfo *statusInfo) {
-	switch key {
-	case "Name":
-		sInfo.name = strings.Trim(value, " \t")
-	case "State":
-		sInfo.status = value[0:1]
-	case "Uid":
-		sInfo.uids = make([]int32, 0, 4)
-		for _, i := range strings.Split(value, "\t") {
-			v, err := strconv.ParseInt(i, 10, 32)
-			if err == nil {
-				sInfo.uids = append(sInfo.uids, int32(v))
+func (p *probe) parseStatusKV(key, value []byte, sInfo *statusInfo) {
```

Comment:
any chance we can get some comments in this function? Mostly to explain the use of byte arrays rather than strings so future maintainers will understand what's going on 😄

Reply:
Addressed in [2f8bcc0](https://github.com/DataDog/datadog-agent/pull/20559/commits/2f8bcc0c68a54294aa7a76c89ca3d0eed086da51)